### PR TITLE
[cmake][addon-devkit] cmake fixes to bring back the Windows build

### PR DESCRIPTION
**Pull request by AchimTuran**
After https://github.com/xbmc/xbmc/pull/9659 went in my binary add-on Windows build was broken. These are the fixes to properly build my AudioDSP add-ons.

@AlwinEsch 
Furthermore it adds the header files to show them in VS.

### CODE REVIEW

**Comment by AchimTuran** [comment]
jenkins build this please

File: project/cmake/addons/CMakeLists.txt [hunk 1]
```diff
@@ -228,10 +228,22 @@ endif()
...
+  set(KODI_ADDON_DEV_KIT_BUILD_ARGS ${BUILD_ARGS})
+  set(__C_FLAGS "${CMAKE_C_FLAGS} ${ARCH_DEFINES}")
+  set(__CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ARCH_DEFINES}")
+  list(APPEND KODI_ADDON_DEV_KIT_BUILD_ARGS -DCMAKE_C_FLAGS=${__C_FLAGS}
+                                            -DCMAKE_CXX_FLAGS=${__CXX_FLAGS})
+else()
+  set(KODI_ADDON_DEV_KIT_BUILD_ARGS ${BUILD_ARGS})
+endif()
+
 ExternalProject_Add(kodi-addon-dev-kit
```
Comment (AchimTuran) [inline project/cmake/addons/CMakeLists.txt]: @fetzerch 
Is there a better way to add kodi-addon-dev-kit without the need of `ExternalProject_Add(...)`

**Comment by AchimTuran** [comment]
jenkins build addons please

File: project/cmake/addons/CMakeLists.txt (hunk 1 above)
Comment (wsnipex) [inline project/cmake/addons/CMakeLists.txt]: did you try add_subdirectory? it has an option to specify the binary dir, which might be needed here.

Comment (AchimTuran) [inline project/cmake/addons/CMakeLists.txt]: Not yet, because I waited for response of @FernetMenta and others in https://github.com/xbmc/xbmc/pull/9843.

**Comment by AchimTuran** [comment]
@AlwinEsch 
I will PR the relevant changes against our repository, when we revert #9843.

Furthermore I close this PR through the discussion in #9843.

**Pull request closed without merging**